handlers/code.go

```diff
@@ -28,28 +31,85 @@ import (
 	"github.com/alecthomas/chroma/lexers"
 	"github.com/alecthomas/chroma/styles"
 	"github.com/dgageot/demoit/files"
+	"gopkg.in/src-d/go-git.v4"
+	"gopkg.in/src-d/go-git.v4/plumbing"
+	"gopkg.in/src-d/go-git.v4/plumbing/object"
 )
 
 // Code returns the content of a source file.
 func Code(w http.ResponseWriter, r *http.Request) {
 	filename := strings.TrimPrefix(r.URL.Path, "/sourceCode/")
+	hash := r.FormValue("hash")
+
+	var contents []byte
+
+	if len(hash) > 0 {
+
+		repoPath, insidePath, err := findRepositoryOfFile(filename)
+		if err != nil {
+			http.NotFound(w, r)
+			return
+		}
+
+		repo, err := git.PlainOpen(repoPath)
```

Comment:
Can you extract those lines to a function so that you only have to `http.Error` once.

Reply:
done